addons/binding/org.openhab.binding.mcp23017/ESH-INF/thing/thing-types.xml

```diff
@@ -0,0 +1,134 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<thing:thing-descriptions bindingId="mcp23017" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+	xmlns:thing="http://eclipse.org/smarthome/schemas/thing-description/v1.0.0"
+	xsi:schemaLocation="http://eclipse.org/smarthome/schemas/thing-description/v1.0.0 http://eclipse.org/smarthome/schemas/thing-description-1.0.0.xsd">
+
+	<thing-type id="mcp23017switch">
```

Comment:
This looks to me like the wrong level of modelling. Your "thing" really is the mcp23017. This physical thing has multiple channels, which are its different pins.

Reply:
@kaikreuzer could you answer the question of @aogorek

Reply:
Oops, sorry - this response was in the middle of my summer vacation, so I seem to have missed that...

@aogorek Yes, 16+16 channels would be the best. What exactly does the "default state" do?

Reply:
@kaikreuzer I think `default state` is the default state when device is powered up. If You connect relays to MCP23017 (that's what I plan to do) and You have active low relays then setting the default state to 'HIGH' will set them off by default. Not sure if this is the best way, because I'm new to OpenHab, but I guess this was @aogorek intention.

Reply:
Thanks for the explanation @Misiu. I think the most convenient way to support this would be as a config parameter on the channel then.

Reply:
@kaikreuzer one more question: do we need 16+16 channels? I'm aware that MCP23017 can have input and output channels (DIGITAL_INPUT and DIGITAL_OUTPUT) but wouldn't it be better to have only 16 channels and have parameter named `type`? I'm new to OpenHab so if I'm wrong then sorry for that.

Reply:
I was just reading up the data sheet and came to the same conclusion: The chip physically only has 16 channels, so yes, we should not add 32 on the Thing, but rather 16, which are then configured to be either input or output.

Reply:
I recently started some refactoring, so probably I'll commit changes soon.
I'm not sure if it will work ok, but I'm planning to do two channel groups,
something like this (in "pseudo xml"):

<channel_group name="input">
   <channel name="A0"/>
   ....
   <channel name="A7"/>
   <channel name="B0"/>
   ....
   <channel name="B7"/>
</channel_group>

<channel_group name="output">
   <channel name="A0"/>
   ....
   <channel name="A7"/>
   <channel name="B0"/>
   ....
   <channel name="B7"/>
</channel_group>

so we will have two channel groups and 16 channels.

What do You thing about it?

Reply:
@aogorek as @kaikreuzer pointed maybe 2x16 channels aren't needed. MCP23017 has only 16 channels so maybe You could add 16 channels but with `type` property (input or output).
In my case I'd like to use that chip with 16 relays, so having properties like standard GPIO's would be awesome:
`gpio="pin:PIN_NUMBER [debounce:DEBOUNCE_INTERVAL] [activelow:yes|no] [force:yes|no] [initialValue:high|low]"`
especially `activelow` and `initialValue`.

what do You think? Have You used it with relays before?